edi_oca/models/edi_exchange_record.py

```diff
@@ -599,7 +626,37 @@ def check_access_rule(self, operation):
 
     def write(self, vals):
         self.check_access_rule("write")
-        return super().write(vals)
+        if "res_id" not in vals and "model" not in vals:
+            return super().write(vals)
+        # Updating main record information
+        res_id_updated = bool("res_id" in vals)
```

Comment:
```suggestion
        res_id_updated = "res_id" in vals
```
it's already a boolean